squire_lib/src/players/mod.rs

```diff
@@ -45,6 +47,15 @@ pub struct Player {
 }
 
 impl Player {
+    /// Returns a string of a player name and, game name
+    pub fn all_names(&self) -> String {
```

Comment:
This is not required as the performance impact is minimal, but it's a good lesson in the power of Rust's type system. This method could return a `Cow<'_, str>` instead of a full String. A Cow is a "copy on write" data structure that either points to an existing value (in this case a string slice) or an owned value. This is particularly helpful to avoid additional allocations.

Here, the `self.name.clone()` branches would be replaced with `Cow::Borrowed(self.name.as_str())` and the `format!` branch would be a `Cow::Owned`.